app_dart/lib/src/request_handlers/presubmit_luci_subscription.dart

```diff
@@ -119,8 +119,29 @@ class PresubmitLuciSubscription extends SubscriptionHandler {
     } else {
       ciYaml = await scheduler.getCiYaml(commit);
     }
-    final Target target = ciYaml.presubmitTargets.where((element) => element.value.name == builderName).single;
+
+    Target target;
+    try {
+      target = ciYaml.presubmitTargets.where((element) => element.value.name == builderName).single;
+    } on Exception {
+      log.warning('Did not find builder with name: $builderName in ciYaml for ${commit.sha}');
+      final List<String> availableBuilderList = ciYaml.presubmitTargets.map((Target e) => e.value.name).toList();
+      log.warning('ciYaml presubmit targets found: $availableBuilderList');
+      rethrow;
+    }
+
+    // alternative solution, just not reschedule.
+    // This might be allowable. We are checking tip of tree but I don't think the
+    // target has been deleted.
+
+    // if (!ciYaml.presubmitTargets.any((element) => element.value.name == builderName)) {
```

Comment:
Let's go with this solution, and add logs here. This way pub/sub messages will block builds update.

Reply:
Okay no problem. I will add it.